inference-engine/src/vpu/graph_transformer/src/model/model.cpp

```diff
@@ -2128,6 +2146,14 @@ void ModelObj::removeUnusedData(const Data& data) {
         _allocator.setNeedToAllocNonIntermData();
     }
 
+    if (const auto dataToShapeEdge = data->_parentDataToShapeEdge) {
+        const auto& shape = dataToShapeEdge->parent();
+        disconnectDatas(dataToShapeEdge);
+        VPU_INTERNAL_CHECK(!shape->childDataToShapeEdges().empty() || !shape->consumerEdges().empty(),
+                "Removed unused data (with name {}) must have a shape data (with name {}) which is a shape "
+                "for other data or has consumer", data->name(), shape->name());
```

Comment:
Will be `shape` dangling reference here because of `disconnectDatas`?

Reply:
It didn't produce such an error, but it's possible. Fixed